src/util/renderers/article.renderer.ts

```diff
@@ -1,13 +1,40 @@
-import domino from 'domino'
 import { MWCapabilities } from '../../Downloader.js'
 import { Dump } from '../../Dump.js'
-import { DELETED_ARTICLE_ERROR } from '../const.js'
-import * as logger from '../../Logger.js'
-import { getStrippedTitleFromHtml } from '../misc.js'
+
+import { Renderer } from './abstractRenderer.js'
+import { ParsoidHtmlRestApiRenderer } from './parsoidHtmlRestApi.renderer.js'
+import { VisualEditorRenderer } from './visualEditor.renderer.js'
+
+export interface MWRendererArgs {
+  mwType: string
+  data: any
+  articleId: string
+  dump: any // TODO: Fix type of dump
+  articleDetailXId: RKVS<ArticleDetail>
+  capabilities: MWCapabilities
+  articleDetail: any
+  isMainPage: boolean
+}
 
 export class ArticleRenderer {
+  private createRenderer(mwRendererArgs: MWRendererArgs): Renderer {
```

Comment:
We probably need multiple ways to create a render:
* full auto-mode (like currently, it decides automatically)
* Get a Desktop or a Mobile (this is needed for the mainpage which is always renderer with dekstop)
* For using an specific API (this is for sure needed with the tests)

Reply:
> Get a Desktop or a Mobile (this is needed for the mainpage which is always renderer with desktop)
I think we can start to work on it (along with other requirements mentioned here) along Wikimedia mobile renderer.

Reply:
Upd:
This is implemented in `RendererBuilder` now.